### PR TITLE
Decreased Docker Image build

### PR DESCRIPTION
**Pull request by TeaDrinkingProgrammer**
First of all, I want to say that this project looks really good: I wanted to make a Rust musicbot myself but when I saw this I thought: ¨this is exactly how I would do it if I had the time (and Rust know-how)". I thus started to look at what I could contribute and began with the Dockerfile, as it seemed to me that it could be smaller. It turns out it can be smaller!

For future reference,I tried:

- Alpine -> too much of a hassle to compile Rust to musl instead of glibc
- Compiling FFMPEG myself -> Makes a small image, but I am no expert at compilation soo to much of a risk of breaking in the future

In the end I settled on these changes:
- Using bitnami/minideb as a base. It turns out installing Python on this image makes it smaller than using the official python:3-slim image.
- Removing unnecessary Python build and develop dependencies
- Using the [static FFMPEG build from yt-dlp](https://github.com/yt-dlp/FFmpeg-Builds). Has the additional benifit of containing some yt-dlp specific fixes. Also offloads the compilation responsibility to them and is assured to work.

Size comparisons (no --squash used):
- Current image from ghcr.io: 631.05 MB
- New image: 248.48 MB

As a side-note, the version with the custom-compiled FFMPEG, came in at 229.4 MBs. The final image might be smaller or even bigger, because the image did not work yet, but I am not going to all that effort to safe 20-40 MBs.

## Notes:
Adding the squash flag made no difference, as the image has very few layers to begin with.  
The new image uses a staticly compiled FFMPEG, which makes architecture dependent. yt-dlps ffmpeg is compiled to both arm64 and amd64 though, so I included an if statement to download the correct version. This should make the new image compatible with both versions, which I think are the only architectures people are using realistically anyways.
**This does mean that the new docker image on ghcr.io should change to be multi-platform, something [in this article](https://callistaenterprise.se/blogg/teknik/2017/12/28/multi-platform-docker-images/)**

### CODE REVIEW

**Comment by TeaDrinkingProgrammer** [comment]
I tried building for arm on my pc but it gave an error on the first cargo build step, but this is likely a cross-compilation problem, not a problem with the new FFMPEG workflow

**Review by jontze — CHANGES_REQUESTED** [review]
LGTM! :rocket: 
Thank you for your contribution and I'm happy to hear that  people are using and enjoining the bot!

The big docker image was always something that annoyed me. So thanks for taking care about this. I wasn't aware of bitnami/minideb, so nice learning for me. :smile: 

Only one tiny styling thing:
I try to stick to [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) with my commit messages in this repo. So it would be really nice if you reword your commit message to something like `chore(docker): ...`

But otherwise LGTM!

File: README.md [hunk 1]
```diff
@@ -32,6 +44,7 @@ docker pull ghcr.io/jontze/cadency_rs:main
 1. Install the latest stable rust toolchaun: https://www.rust-lang.org/tools/install
 2. Intall [yt-dlp](https://github.com/yt-dlp/yt-dlp#installation)
 3. Clone the repository
-4. Create a discord bot so you can run the bot local on your computer
+4. Install [ffmpeg](https://ffmpeg.org/). It is reccomended you install the [yt-dlp build](https://github.com/yt-dlp/FFmpeg-Builds), as this is used in the Docker image as well and includes additional bug-fixes.
```
Comment (jontze) [inline README.md:47]:
```suggestion
4. Install [ffmpeg](https://ffmpeg.org/). It is recomended you install the [yt-dlp build](https://github.com/yt-dlp/FFmpeg-Builds), as this is used in the Docker image as well and includes additional bug-fixes.
```

**Review by jontze — APPROVED** [review]
LGTM!

Just one small typo and we're ready to go, approving already.

**Comment by TeaDrinkingProgrammer** [comment]
Apparently I had to merge the readme, all ready now

**Pull request merged**